src/nfv_test_api/v2/services/gnodeb.py

```diff
@@ -223,28 +223,39 @@ def node_status(self, identifier: str) -> Dict[str, Any]:
         # make sure the config exists
         gnb = self.get_one(identifier)
 
-        if identifier not in self.process_handler.processes:
-            raise NotFound(f"No gNodeB process found for nci {identifier}")
-
-        status: Dict[str, Any] = {}
+        status: Dict[str, Any] = {"pid": None, "status": {}}
         node_name = f"UERANSIM-gnb-{int(gnb.mcc)}-{int(gnb.mnc)}-{int(gnb.nci[:-1], base=0)}"  # type: ignore
         command = [
             "nr-cli",
             node_name,
             "--exec",
             "status",
         ]
+        if identifier not in self.process_handler.processes:
+            raise NotFound(f"No gNodeB process found for nci {identifier}")
 
-        status["pid"] = self.process_handler.processes[identifier].pid
-        stdout, stderr = self.host.exec(command)
-        if stderr:
-            raise NotFound(f"Failed to fetch gNodeB status: {stderr}")
+        errors = []
+        return_code = self.process_handler.processes[identifier].poll()
+        if return_code is not None:
+            # If the process is still in self.process_handler but has terminated then it is a zombie process
+            self.process_handler.kill(identifier)  # handle zombie process
+            errors = [
+                f"The gnodeB process failed with return code {return_code}.",
+                "Killing zombie process, next call to status will raise Not Found.",
+            ]
+        else:
+            # Fetch gnodeB client status only if it is still running
+            status["pid"] = self.process_handler.processes[identifier].pid
+            stdout, stderr = self.host.exec(command)
+            if stderr:
+                raise NotFound(f"Failed to fetch gNodeB status: {stderr}")
```

Comment:
Why is this a NotFound error?

Reply:
Because usually when the status command fail, it means you provided wrong identifier. But I can agree that it could be caused by something else (even if I never encounter them at this point).

Reply:
Maybe then you can just return stderr with the logs, like you do with the zombie process?  This way you can refactor further by:
1. Read the logs
2. If zombie process: return with logs
3. Get status by executing the command
4. If command failed, return with logs
5. Return with status

This way you have never an if in another one

Reply:
Good idea, it made the code much cleaner.